File: typescript-examples/auth-with-secret-otp/utils/typesAndSchemas.ts
import { z } from "zod";

export const createAuthSessionParams = z.object({
  username: z.string().email("Valid email is required"),
  password: z.string().min(1, "Password is required"),
  secret: z.string().min(1, "Secret is required"),
});

export interface Contract {
  id: string;
  name: string;
  supplierName: string;
  supplierPhoneNumber: string;
  effectiveDate: string;
  expirationDate: string;
  state: string;
  detailsUrl: string;
}

File: python-examples/auth-with-secret-otp/utils/types_and_schemas.py
from pydantic import BaseModel, EmailStr, Field


class CreateAuthSessionParams(BaseModel):
    username: EmailStr
    password: str = Field(min_length=1, description="Password is required")
    secret: str = Field(min_length=1, description="Secret is required")


class Contract(BaseModel):
    id: str
    name: str
    supplier_name: str
    supplier_phone_number: str
    effective_date: str
    expiration_date: str
    state: str
    details_url: str
